test/hotspot/jtreg/containers/docker/TestMemoryWithCgroupV1.java

```diff
@@ -81,10 +81,17 @@ private static void testMemoryLimitWithSwappiness(String dockerMemLimit, String
         Common.addWhiteBoxOpts(opts);
 
         OutputAnalyzer out = Common.run(opts);
-        out.shouldContain("Memory and Swap Limit is: " + expectedReadLimit)
+        // in case of warnings like : "Your kernel does not support swap limit
+        // capabilities or the cgroup is not mounted. Memory limited without swap."
+        // we only have Memory and Swap Limit is: <huge integer> in the output
+        try {
+            out.shouldContain("Memory and Swap Limit is: " + expectedReadLimit)
                 .shouldContain(
                         "Memory and Swap Limit has been reset to " + expectedResetLimit + " because swappiness is 0")
                 .shouldContain("Memory & Swap Limit: " + expectedLimit);
+        } catch (RuntimeException ex) {
+            out.shouldMatch("Memory and Swap Limit is: [0-9]+");
+        }
```

Comment:
With the `catch` clause added, it looks like the tests inside the `try` block don't do anything anymore, even on a host where the kernel actually supports swap limit capabilities.

Maybe the correct way to handle this is: if "Your kernel does not support memory swappiness capabilities or the cgroup is not mounted." is in the output, then skip the test inside the `try` block?

Reply:
I don't know why that wasn't liked. Your patch essentially reduces `testMemoryLimitWithSwappiness` to test for only this:

```
out.shouldMatch("Memory and Swap Limit is: [0-9]+");
```

So if we go with this way, you may as well remove everything inside the `try` block.

Reply:
The issue with asserting a specific error from a container engine is that it might be different from one (`docker`) to the other (`podman`). Another way to address this, could be to pre-assert that on systems where this is happening (no swap), to skip the test. I.e. along the lines of:

```
docker run --memory 100 --memory-swap 200
```
and inside a container have this code:

```
if (metrics.getMemoryAndSwapLimit() == metrics.getMemoryLimit()) {
   // system doesn't support swap, skip test
}
```

Basically akin to: https://github.com/openjdk/jdk/blob/master/test/lib/jdk/test/lib/containers/cgroup/MetricsTesterCgroupV1.java#L283..L284

Reply:
Looks like changing  GRUB_CMDLINE_LINUX_DEFAULT / GRUB_CMDLINE_LINUX in /etc/default/grub file
by adding   cgroup_enable=memory swapaccount=1
can solve the issue on Linux ppc64le machines (see https://www.suse.com/support/kb/doc/?id=000018585 ).
I am still not sure why this was not needed on our Linux x86_64 machines.

Without this added setting, 'docker info' showed :
WARNING: No swap limit support

With this setting, the warning is gone.

Reply:
I think the error message is not very clear. How about 

```
You may need to add "cgroup_enable=memory swapaccount=1" to the Linux kernel boot parameters.
```